src/NuGet.Clients/NuGet.VisualStudio.Common/ServiceLocator.cs

```diff
@@ -117,27 +113,38 @@ public static void InitializePackageServiceProvider(IAsyncServiceProvider provid
                 }
             }
 
-            return Package.GetGlobalService(typeof(TService)) as TInterface;
+            return null;
         }
 
-        private static async Task<TService> GetDTEServiceAsync<TService>() where TService : class
+        /// <summary>
+        /// Fetches a MEF registered service if available.
+        /// This method should be called from a background thread only. 
+        /// </summary>
+        /// <typeparam name="TService"></typeparam>
+        /// <returns>The instance of the service request, <see langword="null"/> otherwise. </returns>
+        /// <remarks>
+        /// This method should only be preferred when using MEF imports is not easily achievable.
+        /// Prefer this over <see cref="GetInstanceAsync{TService}"/> when the service requesting a MEF service.
+        /// A good rule of thumb is that internal NuGet services should call this method over <see cref="GetInstanceAsync{TService}"/>.
```

Comment:
Avoid using the term "rule of thumb" due to its origin. Use "general rule" instead.

Reply:
Welp, I had no clue about that particular origin story. 

TIL!

Thanks!